### DISCUSSION TITLE
Can i load large datasets from postgres to Qfield to use as a viewer?

### DISCUSSION THREAD

**Discussion by duranduran127** [opengisch/QField#4921] [Q&A]
I currently use ESRI field maps to display large datasets for our field guys to use while they are driving around such as parcels with information, contours etc. All are large datasets that are costly to store in ESRI and I was wondering if Qfield has the ability to display datasets from postgres databases? I have experimented before with Qfield and trying to connect to postgres and even upgrading to the pro version of the cloud to try to make it work with no help. I understand Qfield has some sort of 1GB limit but im not sure if that pertains to if you are pulling data from a database like postgres or if that is just direct storage limits. I have about 20GB of data in total (could be more) in my QGIS project. My goal is for my field guys to be able to use Qfield as a viewer that would display information, has anyone figured out how to use large datasets from postgres in Qfield?

**Comment by m-kuhn** [comment]
Hi.

QField as well as QFieldCloud can talk to the cloud.

However, getting 20GB through QFieldCloud when packaging a project takes a long time and is likely to hit the 10min time maximum for jobs.

If your data is static, you could upload a copy of it as gpkg to QFieldCloud and choose [copy as layer action](https://docs.qfield.org/get-started/tutorials/get-started-qfs/#layer-configuration). NB; there is no 1 GB limit.

You could also consider to have multiple projects that only work on filtered regions (which are smaller).

If you are sure you always have network coverage, you can also let QField directly talk to the postgres database, by choosing the [No Action](https://docs.qfield.org/get-started/tutorials/get-started-qfs/#layer-configuration) layer action, you just need to make sure that QField has the permissions to access.
